### PR TITLE
Add: typing field_conversions & unithelper modules

### PR DESCRIPTION
**Pull request by enadeau**
More works on #373

## Description

It done!! I've added type annotation to the last file from the library!

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [x] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

### CODE REVIEW

**Comment by enadeau** [comment]
There is some issue with python 3.8 and pint. The latest release of pint is 0.22 and includes nice typing improvement but is doesn't support python 3.8 (see changelog [here](https://github.com/hgrecco/pint/blob/2852f36db9aab916dbaaa68232ef45efccac3426/CHANGES#L15C4-L15C4))

I'll need to find a fix for that unless we decide to follow the same logic as they do and remove support for 3.8

File: changelog.md [hunk 1]
```diff
@@ -4,12 +4,12 @@
 
 - Apply flake8 and numpy docstrings to limiter & protocol (@lwasser, #326)
 - Update client's stream method to warn when using unofficial parameters (@enadeau, #385)
-- Type annotation to client file (@enadeau, #384)
 - Fix docstring in SleepingRateLimitRule (@enadeau)
 - Fix: rename `SubscriptionCallback.validate` -> `SubscriptionCallback.validate_token` to avoid conflict with `pydantic.BaseModel` (@lwasser, #394)
 - Fix: docstrings in model.py, documentation errors, findfonts warning suppression by removing opengraph (temporarily), typing updates (@lwasser, #387)
 - Fix: read the docs is breaking due to pydantic json warnings, also update python version on build and sync pr previews (@lwasser, #412)
 - Fix: update master to main in all builds (@lwasser)
+- Type annotation for all file in the library (@enadeau, #384, #415)
```
Comment (lwasser) [inline changelog.md:12]: 🎉 🚀

**Comment by lwasser** [comment]
> There is some issue with python 3.8 and pint. The latest release of pint is 0.22 and includes nice typing improvement but is doesn't support python 3.8 (see changelog [here](https://github.com/hgrecco/pint/blob/2852f36db9aab916dbaaa68232ef45efccac3426/CHANGES#L15C4-L15C4))

@enadeau this is super timely - [i created this issue](https://github.com/stravalib/stravalib/issues/301) in the fall about python support. Essentially we could decide to support NEP-29 / scientific python loosely. i think if we did we'd also be dropping 3.9 soonish if not already BUT we could just decide to drop 3.8 now given a lot of the (scientific) ecosystem is doing that. let's chat in that issue about our policy around this. i'm happy to update docs once we decide.

File: changelog.md (hunk 1 above)
Comment (lwasser) [reply changelog.md:12]: amazing work! i haven't looked at the files yet just responding to the NEP-29 / pint / python 3.8 comment below!!

**Pull request merged**